File: SNS.Web/src/app/jobs/company-create-requests/contracts/company-create-request-details.dto.ts
import { ProfileSnapshotDto } from '../../../profiles/profiles/contracts/profile-snapshot.dto';
import { CompanyCreateRequestStatus } from '../../enums/company-create-request-status.enum';

export interface CompanyCreateRequestDetailsDto {
    id: string;
    profileId: string;
    profile: ProfileSnapshotDto;
    name: string;
    industry: string;
    websiteUrl: string | null;
    logoUrl: string | null;
    status: CompanyCreateRequestStatus;
    createdCompanyId: string | null;
    reviewedByProfileId: string | null;
    reviewNote: string | null;
    createdAt: string;
    reviewedAt: string | null;
}

File: backend/SNS.Application/Jobs/CompanyCreateRequests/Contracts/CompanyCreateRequestDetailsDto.cs
using SNS.Application.Profiles.Profiles.Contracts;
using SNS.Domain.Jobs.Enums;

namespace SNS.Application.Jobs.CompanyCreateRequests.Contracts;

public sealed record CompanyCreateRequestDetailsDto(
    Guid Id,
    Guid ProfileId,
    ProfileSnapshotDto Profile,
    string Name,
    string Industry,
    string? WebsiteUrl,
    string? LogoUrl,
    CompanyCreateRequestStatus Status,
    Guid? CreatedCompanyId,
    Guid? ReviewedByProfileId,
    string? ReviewNote,
    DateTime CreatedAt,
    DateTime? ReviewedAt
);
